### DISCUSSION TITLE
Clarification on Retrieving Anonymized Buyer Information Without Restricted Data Token (RDT) from an Order

### DISCUSSION THREAD

**Discussion by brunoamorimdev** [amzn/selling-partner-api-models#4147] [Q&A]
Hi,

I need to associate a retrieved order with its buyer without accessing any restricted data. After reviewing the documentation, it appears that I can obtain an anonymized email using the [getOrderBuyerInfo](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#getorderbuyerinfo) operation. However, I’m concerned about whether it’s necessary to generate a Restricted Data Token (RDT) to perform this request in production.

The documentation doesn’t clearly indicate whether an RDT is required for retrieving the anonymized email of the buyer. Could you please clarify if generating an RDT is necessary for this operation, or if the anonymized email can be obtained without it?

Thank you for your assistance.

Best regards,
Bruno

**Comment by jawheral** [comment]
Hi @brunoamorimdev,

When retrieving orders using [getOrder](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#get-ordersv0ordersorderid) for example you will find the BuyerInfo as part of the [Order](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#order) in the response with the BuyerEmail (anonymised) available in the response. Other restricted data under BuyerInfo such as BuyerName or BuyerTaxInformation will require the right RDT to be retrieved.

The [getOrderBuyerInfo](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#getorderbuyerinfo) should provide similar data.


Best,
Jawher
Selling Partner API Solutions Architect

**Comment by brunoamorimdev** [comment]
Hi @jawheral,

Got it, I see that de Order definition shows that BuyerInfo wild be returned, but looking for an example in the Use Case Guide the payload does not return the BuyerInfo completely, just an empty object, so I guessed it was just available through the /buyerInfo endpoint or making the same API call with the RDT.

Best regards,
Bruno
